### DISCUSSION TITLE
How to keep the alpha channel after format a gif image to a webp image?

### DISCUSSION THREAD

**Discussion by ghost** [ImageMagick/ImageMagick#6618] [Help]
### ImageMagick version

7.1.0-52

### Operating system, version and so on

Debian 6.3.0-18+deb9u1

### Description

I try to convert a gif file to webp format, but if the gif have some transparent part, my result will loss it, how can I keep the alpha channel as input after format?

the api i use:
```
// Sets the format of a particular image in a sequence.
//
// format: the image format.
//
func (mw *MagickWand) SetImageFormat(format string) error {
	csformat := C.CString(format)
	defer C.free(unsafe.Pointer(csformat))
	ok := C.MagickSetImageFormat(mw.mw, csformat)
	return mw.getLastErrorIfFailed(ok)
}
```

**Comment by snibgo** [comment]
At the command line, IM writes to webp with transparency, with no problem.

If you are having problems writing code to do the same thing, I suggest you show the complete program you have written so far. Also please state what language you are using (C, C++, whatever).

I suggest you also paste your program so that it doesn't get re-formatted by the forum software.

**Reply by ghost** [reply]
We use Golang, and this is the function we used to do the convert

```
func toWEBP(mw *imagick.MagickWand) error {
	mw.ResetIterator()
	defer mw.ResetIterator()
	frames := int(mw.GetNumberImages())
	mw.ResetIterator()
	
	for i := 0; i < frames; i += 1 {
		mw.SetIteratorIndex(i)
		if err := mw.SetImageIterations(0); err != nil {
			return err
		}
		if err := mw.SetImageFormat("WEBP"); err != nil {
			return err
		}
		w, h, x, y, err := mw.GetImagePage()
		if err != nil {
			return err
		}
		if err = mw.ResetImagePage(fmt.Sprintf("%dx%d+%d+%d", w, h, x, y)); err != nil {
			return err
		}
	}
	
	return nil
}
```

**Comment by fmw42** [comment]
Perhaps you need to update your webp. Identify your version of webp.

**Comment by snibgo** [comment]
> We use Golang, and this is the function we used to do the convert

I don't use Golang, but I can't see anything in that code that writes the images to a file.
